### PR TITLE
add basic spinner and loading button

### PR DESCRIPTION
**Pull request by rick-gnous**
Just to test if it's work. I'll add more options after, based on patternfly documentation.

### CODE REVIEW

File: src/spinner.rs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+pub enum SpinnerSize {
+    Sm,
+    Md,
+    Lg,
+    Xl,
+}
+
+impl Default for SpinnerSize {
+    fn default() -> Self {
+        Self::Xl
```
Comment (ctron) [inline src/spinner.rs:13]: Looking and the docs, and I might be wrong, but it looks like the "default" is actually without a size class.

**Comment by ctron** [comment]
I really appreciate the effort, implementing the "legacy" part too. However (unless you do need it) I would suggest to drop it.

I done so, the be easy to change the `SpinnerSize` enum into e.g.:

```rust
pub struct SpinnerSize {
  Sm,
  Md,
  Lg,
  Xl,
  Custom(u16),
}
```

Or using `Custom(String)` to allow specifying the CSS unit too.

**Comment by ctron** [comment]
> 

True, that seems out of sync. I peeked the rendered code: ReactJS has valuetext, while HTML has label.

:shrug: Just do what you think is best.

**Comment by rick-gnous** [comment]
> I really appreciate the effort, implementing the "legacy" part too. However (unless you do need it) I would suggest to drop it.

I've implemented the legacy part for the loading buttons. If I use svg for spinner, it's different:

**Legacy**
legacy

**SVG**
svg

I don't know which visual you prefer.

> I done so, the be easy to change the `SpinnerSize` enum into e.g.:
> 
> ```rust
> pub struct SpinnerSize {
>   Sm,
>   Md,
>   Lg,
>   Xl,
>   Custom(u16),
> }
> ```
> 
> Or using `Custom(String)` to allow specifying the CSS unit too.

oh, you're right! I'll change it for the CSS unit.

File: src/spinner.rs (hunk 1 above)
Comment (rick-gnous) [reply src/spinner.rs:13]: Yes you're right, I've not noticed that. Thank!

**Comment by rick-gnous** [comment]
> True, that seems out of sync. I peeked the rendered code: ReactJS has valuetext, while HTML has label.
> 
> shrug Just do what you think is best.

I think I'll keep label. Maybe change it later if devs don't like it.

**Comment by ctron** [comment]
Regarding legacy vs svg. I don't have a real preference. Focusing on "svg" would make the code simpler though. So I think that would be an argument for dropping "legacy".

But I will leave the decision to you, as you are the one doing all the work.

**Comment by jbtrystram** [comment]
It looks like the "legacy" turns the wrong way somehow

**Comment by rick-gnous** [comment]
As all the examples in the patternfly doc use svg, I'll drop the "legacy".


> It looks like the "legacy" turns the wrong way somehow

Yes, i don't really know why. It turns the right way in the doc.

**Comment by ctron** [comment]
@rick-gnous just let me know when you think this is good to be merged

**Comment by rick-gnous** [comment]
I think it's good to be merged. I'll open a new PR if I add new components.

**Review by ctron — APPROVED** [review]

**Pull request merged**